charts/orchestrator/templates/_cr_lifecycle_helper.tpl

```diff
@@ -0,0 +1,142 @@
+{{- define "manage-cr-lifecycle-on-action" }}
+  {{- $resourceAPIGroup := printf "%s.%s" .kinds .apiGroup }}
+  {{- $releaseNameKind := printf "%s-%s" .release.Name .kind |lower }}
+  {{ if or .isEnabled (and (not .isEnabled) (and .hasCRDInstalled (not (empty (lookup (printf "%s/%s" .apiGroup .groupVersion) .kind (dig "targetNamespace" "" . ) .resourceName ))))) }}
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ $releaseNameKind }}
+  namespace: {{ .release.Namespace }}
+  annotations:
+    "helm.sh/hook": pre-delete,{{ if .isEnabled }}post-install,post-upgrade,post-rollback{{ else }}pre-upgrade,pre-rollback{{ end }}
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded,hook-failed
+    "helm.sh/hook-weight": "-1"
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ $releaseNameKind }}
+  annotations:
+    "helm.sh/hook": pre-delete,{{ if .isEnabled }}post-install,post-upgrade,post-rollback{{ else }}pre-upgrade,pre-rollback{{ end }}
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded,hook-failed
+    "helm.sh/hook-weight": "0"
+rules:
+  - apiGroups:
+    - apiextensions.k8s.io
+    resources:
+    - customresourcedefinitions
+    verbs:
+    - get
+  - apiGroups:
+    - {{ .apiGroup }}
+    resources:
+    - {{ .kinds |lower}}
+    verbs:
+    - get
+    - list
+    - delete
+    - patch
+    - update
+    - create
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: {{ $releaseNameKind }}
+  annotations:
+    "helm.sh/hook": pre-delete,{{ if .isEnabled }}post-install,post-upgrade,post-rollback{{ else }}pre-upgrade,pre-rollback{{ end }}
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded,hook-failed
+    "helm.sh/hook-weight": "0"
+subjects:
+  - kind: ServiceAccount
+    name: {{ $releaseNameKind }}
+    namespace: {{ .release.Namespace }}
+roleRef:
+  kind: ClusterRole
+  name: {{ $releaseNameKind }}
+  apiGroup: rbac.authorization.k8s.io
+    {{- if .isEnabled }}
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ trunc -57 (printf "%s-%s" $releaseNameKind (.release.IsInstall | ternary "install" "upgrade" ) ) }} # Fixes https://github.com/parodos-dev/orchestrator-helm-chart/issues/160
```

Comment:
just to confirm: we check for `isInstall` to identify the first time the chart is installed, while "upgrade" should satisfy both `upgrade` and `rollback` scenario?

Reply:
Yes.